project-base/storefront/cypress/e2e/cart/cartPage.cy.ts

```diff
@@ -1,63 +1,154 @@
+        takeSnapshotAndCompare('increase-and-decrease-cart-item-quantity-using-slow-spinbox-click_decrease');
     });
 
     it('should remove products from cart', () => {
         cy.getByTID([[TIDs.pages_cart_list_item_, 0], TIDs.pages_cart_removecartitembutton]).click();
-        cy.getByTID([[TIDs.pages_cart_list_item_, 0], TIDs.pages_cart_list_item_name]).contains(
-            products.helloKitty.fullName,
-        );
-        cy.getByTID([[TIDs.pages_cart_list_item_, 1]]).should('not.exist');
-        checkCartTotalPrice('€279.92');
+        takeSnapshotAndCompare('remove-products-from-cart_last_product_left');
 
         cy.getByTID([[TIDs.pages_cart_list_item_, 0], TIDs.pages_cart_removecartitembutton]).click();
-        cy.getByTID([[TIDs.pages_cart_list_item_, 0]]).should('not.exist');
-        cy.getByTID([TIDs.cart_page_empty_cart_text]).should('be.visible');
+        takeSnapshotAndCompare('remove-products-from-cart_empty_cart');
     });
 
-    it('should add and then remove promo code from cart', () => {
+    it('spinbox buttons should not be clickable if they cannot be used due to min/max quantity', () => {
+        cy.getByTID([[TIDs.pages_cart_list_item_, 0], TIDs.forms_spinbox_decrease]).should(
+            'have.css',
+            'pointer-events',
+            'none',
+        );
+        cy.getByTID([[TIDs.pages_cart_list_item_, 0], TIDs.forms_spinbox_increase]).should(
+            'not.have.css',
+            'pointer-events',
+            'none',
+        );
+
+        cy.getByTID([[TIDs.pages_cart_list_item_, 0], TIDs.spinbox_input]).type('10000');
+
+        cy.getByTID([[TIDs.pages_cart_list_item_, 0], TIDs.forms_spinbox_decrease]).should(
+            'not.have.css',
+            'pointer-events',
+            'none',
+        );
+        cy.getByTID([[TIDs.pages_cart_list_item_, 0], TIDs.forms_spinbox_increase]).should(
+            'have.css',
+            'pointer-events',
+            'none',
+        );
+    });
+
+    it('should add promo code to cart, check it, remove promo code from cart, and then add a different one', () => {
         cy.getByTID([TIDs.blocks_promocode_add_button]).click();
         cy.get('#blocks-promocode-input').should('be.visible').type('test', { force: true });
         cy.getByTID([TIDs.blocks_promocode_apply_button]).click();
-        cy.getByTID([TIDs.blocks_promocode_promocodeinfo_code]).contains('test');
-        cy.getByTID([TIDs.pages_cart_cartpreview_discount]).contains('-€27.99');
-        checkCartTotalPrice('€647.53');
+        checkAndHideSuccessToast('Promo code was added to the order.');
+
+        takeSnapshotAndCompare('apply-remove-and-apply-promocode_after-applying-first-promocode');
 
         continueToTransportAndPaymentSelection();
         checkUrl(url.order.transportAndPayment);
+        takeSnapshotAndCompare('transport-and-payment-page-with-applied-promo-code');
+
+        goBackToCartPage();
+        checkUrl(url.cart);
```

Comment:
This style of "do something" and then "check something" for me would be better completely in one function, so it is easily reusable and it is always checked, that it is working correctly.

Reply:
I agree, I try to follow this, but sometimes fail to find the balance. I will keep this in mind. If you have any specific ideas for this PR, let's discuss them.